msdk-id-sirius/src/main/java/SiriusIdentificationMethod.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * (C) Copyright 2015-2017 by MSDK Development Team
+ *
+ * This software is dual-licensed under either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1 as published by the Free
+ * Software Foundation
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by the Eclipse Foundation.
+ */
+
+import de.unijena.bioinf.ChemistryBase.chem.PrecursorIonType;
+import de.unijena.bioinf.ChemistryBase.ms.Ms2Experiment;
+import de.unijena.bioinf.ChemistryBase.ms.Peak;
+import de.unijena.bioinf.ChemistryBase.ms.Spectrum;
+import de.unijena.bioinf.sirius.IdentificationResult;
+import de.unijena.bioinf.sirius.IsotopePatternHandling;
+import de.unijena.bioinf.sirius.Sirius;
+import io.github.msdk.MSDKException;
+import io.github.msdk.MSDKMethod;
+import io.github.msdk.MSDKRuntimeException;
+import io.github.msdk.datamodel.IonAnnotation;
+import io.github.msdk.datamodel.MsSpectrum;
+import io.github.msdk.datamodel.MsSpectrumType;
+import io.github.msdk.datamodel.SimpleIonAnnotation;
+import io.github.msdk.datamodel.SimpleMsSpectrum;
+import io.github.msdk.spectra.centroidprofiledetection.SpectrumTypeDetectionAlgorithm;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import javax.annotation.Nullable;
+import org.openscience.cdk.interfaces.IMolecularFormula;
+import org.openscience.cdk.silent.SilentChemObjectBuilder;
+import org.openscience.cdk.tools.manipulator.MolecularFormulaManipulator;
+
+public class SiriusIdentificationMethod implements MSDKMethod<List<IonAnnotation>> {
+
+  private final Sirius sirius;
+  private MsSpectrum ms1;
+  private MsSpectrum ms2;
+  private double parentMass;
+  private String ion;
+  private int numberOfCandidates;
+  private List<IonAnnotation> result;
+
+  public SiriusIdentificationMethod(@Nullable MsSpectrum ms1, MsSpectrum ms2, double parentMass,
```

Comment:
Please use Nonnull for ms2, and change parentMass from double to @Nonnull Double. 
It is better to avoid using primitive types unless it is for performance reasons (like large data arrays).